src/HospitalModel.H

```diff
@@ -123,6 +123,12 @@ void HospitalModel<PCType, PTDType, PType>::treatAgents (PCType& a_agents, /*!<
                 ParallelForRNG(np, [=] AMREX_GPU_DEVICE (int i, RandomEngine const& engine) noexcept {
                     if (!inHospital(i, ptd)) {
                         // agent is not in hospital
+                        // determine if agent dies anyway.
+                        if (Random(engine) < disease_parm_d->m_unhospToDeath[age_group_ptr[i]]) {
+                            is_alive_ptr[i] = 0;
+                            flag_status_ptr[i] *= -1;
+                            status_ptrs[d][i] = Status::dead;
+                        }
```

Comment:
Given that this is the `HospitalModel`, it may be confusing to have this code here. Why not have it in `DiseaseStatus::updateAgents()` (using, say a `marked_for_death` flag similar to `marked_for_hosp`)?